### PR TITLE
Crossroads Kids' Club Backfill (with Contentful)

### PR DESCRIPTION
**Pull request by liamillard**
### About

This PR adds shovel capability for crossroads kids club data coming from Contentful. This PR _only_ includes the backfill.

The steps are:
Screen Shot 2022-01-11 at 12 13 54 PM

#### `create_categories`
This is all placeholder code. Categories do not yet exist in Contentful, so we're hard-coding them 😅 I hope this step will be easy to update once kids club categories _are_ available in Contentful.

#### `fetch_and_save_assets`
Contentful calls images `Assets`. In this step, we are fetching all of the assets from Contentful. Yes, even assets not associated with a kids' club Entry are synced. Later, unused assets will be deleted.

#### `fetch_and_save_content_items`
This is the core of the data syncing, and a lot of things are happening:
- we are fetching all `Entries` from contentful
- we find what entries are kids club content, based on their category (and match the required distribution channel) and save those to postgres
- we link the item to its asset
- we save the video URL as a row in the `media` table in postgres
- we link the item to its correct category (older kids, or younger kids, based on the title because these aren't setup in Contentful)
- finally, for a backfill, we delete any items that weren't newly synced

#### `remove_unused_assets`
Here we find all assets that aren't linked to a `content_item` and delete them from the database.

#### So why are you syncing assets first?
For the history of this decision, see the discussion on [this to-do in Basecamp](https://3.basecamp.com/3926363/buckets/25558802/todos/4502039686).

### Testing
Before testing, to airflow, add:
- [ ] A connection to the `crossroads_kids_club_apollos_postgres` (credentials from @vinnyjth)
- [ ] A variable called `crossroads_contentful_api_key` (credentials from @liamillard)

1. Start up Airflow, and filter the dags list by `crossroads`.
2. Unpause the `crossroads_kids_club_backfill_contentful_content_item_dag`. Wait for it to complete, it will take a few minutes.
3. Check out all the data in postgres 😅

### Screenshots
Screen Shot 2022-01-11 at 12 13 47 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,63 @@
-# Apollos Shovel 
+# Apollos Shovel
```
Comment (vinnyjth) [inline README.md:1]: 💥 

can you guys come use some of our tech everyday?

Comment (liamillard) [reply README.md:1]: I'm a firm believer in the idea that if you're using something for the first time, you're updating the readme 🙂

**Comment by liamillard** [comment]
This is finally ready for review @vinnyjth! If there are a lot of things wrong with it I'm happy to pair rather than make you write out a million github comments 😅

File: dags/contentful_crossroads/cr_contentful_content_items.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+    def map_content_to_columns(self, entry):
+        fields = self.get_entry_fields(entry)
+        sys = entry.sys
+        return {
+            "created_at": self.kwargs["execution_date"],
+            "updated_at": self.kwargs["execution_date"],
+            "origin_id": sys["id"],
+            "origin_type": "contentful",
+            "apollos_type": "Media",
```
Comment (vinnyjth) [inline dags/contentful_crossroads/cr_contentful_content_items.py:106]:
```suggestion
            "apollos_type": "ContentItem",
```

Comment (vinnyjth) [inline dags/contentful_crossroads/cr_contentful_content_items.py:106]:
```suggestion
            "apollos_type": "MediaContentItem",
```

File: dags/contentful_crossroads/cr_contentful_content_items.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        origin_id = entry.sys["id"]
+        pg_entry_id = self.pg_hook.get_first(
+            f"SELECT id FROM content_item WHERE origin_id = '{origin_id}'"
+        )[0]
+        # Getting the video url is a big ol' TBD.
+        # We don't have streaming links for the video files yet.
+        # In the meantime, we'll use a placeholder, the same for all videos.
+        # fields = self.get_entry_fields(entry)
+        # video_url = fields['file']['url']
+        video_url = "https://player.vimeo.com/external/522397154.m3u8?s=288eb6e0d740e0b2344b7d003b59f1b4245dfde3"
```
Comment (vinnyjth) [inline dags/contentful_crossroads/cr_contentful_content_items.py:125]: We have some vimeo api code hiding somewhere :D

**Review by vinnyjth — CHANGES_REQUESTED** [review]
One teeny thing change and then we are good to Lia!

**Pull request merged**